pymatgen/analysis/phase_diagram.py

```diff
@@ -599,14 +599,98 @@ def get_equilibrium_reaction_energy(self, entry):
             equilibrium reaction energy <= 0.
         """
         if entry not in self.stable_entries:
+            # NOTE Comprhys: could rasing a warning and returning NaN be
+            # a better solution here than raising an Error? an Error interupts a
+            # whole program even if it could complete with a NaN.
             raise ValueError("Equilibrium reaction energy is available only "
                              "for stable entries.")
         if entry.is_element:
             return 0
         entries = [e for e in self.stable_entries if e != entry]
         modpd = PhaseDiagram(entries, self.elements)
-        return modpd.get_decomp_and_e_above_hull(entry,
-                                                 allow_negative=True)[1]
+        return modpd.get_decomp_and_e_above_hull(entry, allow_negative=True)[1]
+
+    def get_decomposition_energy(self, entry):
+        """
+        NOTE a different name might be better to avoid confusion with
+        the decomposition of unstable compounds.
+
+        Provides the decomposition energy of an entry from the neighboring
+        stable entries excluding the entry in question. For unstable entries
+        this is the same as the energy above the hull.
+
+        Args:
+            entry: A PDEntry like object
+
+        Returns:
+            Decomposition energy per atom of entry. Stable entries should have
+            decomposition energies <= 0.
+        """
+        # Handle unstable materials
+        if entry not in self.stable_entries:
+            return self.get_e_above_hull(entry)
+
+        # Handle stable elemental materials
+        if entry.is_element:
+            return 0
+
+        # Handle stable compounds
+        return self.get_decomp_and_decomp_energy(entry)[1]
+
+    def get_decomp_and_decomp_energy(self, entry, space_limit=200):
```

Comment:
Maybe better to rename this (`get_decomposition_products_and_energies` maybe?), method also needs a description in its docstring.

Reply:
Sure, shall I also adopt the same standard of `{c: fractional amt}` for `decomp` as in the e above hull method? Currently it's inconsistent between stable and unstable materials as I make use of the other method